src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -683,7 +714,10 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
     std::unique_lock<std::mutex> lock(state_mutex);
     auto const threads_at = threads.find(thread_id);
     if (threads_at != threads.end()) {
-      threads_at->second.split_and_retry_oom_injected = num_ooms;
+      auto& oom_state      = threads_at->second.split_and_retry_oom;
+      oom_state.hit_count  = num_ooms;
+      oom_state.filter     = oom_type::CPU_OR_GPU;
+      oom_state.skip_count = 0;
```

Comment:
Looks like we've added the some of the code to handle this below.  Maybe we should just do it all in this PR?